### PR TITLE
Implement new endpoint for protected branches

### PR DESCRIPTION
**Pull request by AAraKKe**
This is related to issue #331, I needed to fix it for a project so if you like feel free to take it.

I also modified the tests but I am not an expert on rspec, maybe there is something you would like to adapt on that side. I have tried this and seems to be working for my use case.

### CODE REVIEW

**Comment by AAraKKe** [comment]
I run the tests locally and they run, I am not sure why they are not running on travis.

**Comment by AAraKKe** [comment]
I just noticed an issue, when the new delete method is called no response is sent and the decode method raises an issue, I am not sure the best way to fix it, if you let me know I could fix it before accepting the PR

File: lib/gitlab/request.rb [hunk 1]
```diff
@@ -32,7 +32,8 @@ def self.parse(body)
 
     # Decodes a JSON response into Ruby object.
     def self.decode(response)
-      JSON.load response
+      return JSON.load response if response
+      {}
```
Comment (NARKOZ) [inline lib/gitlab/request.rb]: Can you change this to `response ? JSON.load(response) : {}`?

Comment (AAraKKe) [reply lib/gitlab/request.rb]: I just pushed this modification

**Comment by NARKOZ** [comment]
Thank you.

**Comment by AAraKKe** [comment]
No problem!

**Pull request merged**